source/structured_data.cc

```diff
@@ -855,9 +855,24 @@ namespace aspect
     template <int dim>
     double
     StructuredDataLookup<dim>::get_data(const Point<dim> &position,
-                                        const unsigned int component) const
+                                        const unsigned int component,
+                                        bool crash_if_not_in_range) const
     {
       Assert(component<n_components, ExcMessage("Invalid component index"));
+
+      if (crash_if_not_in_range)
+        {
+          const std::vector<double> x_coordinates = get_interpolation_point_coordinates(0);
```

Comment:
This is expensive because you are copying the whole vector. But you can make it cheaper:
```suggestion
          const std::vector<double> &x_coordinates = get_interpolation_point_coordinates(0);
```
Same for the place a few lines down.